x-pack/plugins/infra/public/hooks/use_readOnly_badge.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '../../../../../src/plugins/kibana_react/public';
+
+export const useReadOnlyBadge = (readOnlyBadge = false) => {
+  const chrome = useKibana().services.chrome;
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Should this be here? (This may well be me missing something but the `exhaustive-deps` rule would usually go above something with dependencies, e.g. `useCallback` or `useEffect`). Out of interest what's the reasoning to make the deps non-exhaustive here?

Reply:
@Kerry350 badge is indeed used inside a `useCallback` and `useEffect`. I grabbed this part of code from the existing `header.tsx` which I plan to remove together with the `Header` component. 

Header component is redundant now that I created these new hooks `use_readonly_badge` and `use_breadcrumb`. The new use_breadcrumb hook is being passed an extra breadcrumb object for the current page, for example `Inventory`, `Metrics Explorer`. This information for the current page was not present where we rendered the Header before, it was outside of the route component if that makes sense. I mention this in case you have any reasons we might want to keep the header component.